src/main/java/com/google/devtools/build/docgen/templates/attributes/common/target_compatible_with.html

```diff
@@ -0,0 +1,75 @@
+<p>
+<code>List of <a href="../build-ref.html#labels">labels</a>; optional;</code>
+</p>
+
+<p>
+A list of
+<code><a href="platform.html#constraint_value">constraint_values</a></code>
+that must be present in the target platform for this target. This is in
+addition to any constraints already set by the rule type. Constraints are used
+to restrict the list of available target platforms. If the target platform does
+not satisfy all listed constraints then the target is considered "incompatible"
+and will be skipped for building and testing.
+</p>
+
+<p>
+Targets that transitively depend on incompatible targets are themselves also
+considered incompatible. These will also be skipped for building and testing.
+</p>
+
+<p>
+An empty list signifies that the target is compatible with all platforms.
+<p>
+
+<p>
+The simplest usage of this attribute restricts a target to a single platform.
+The target will not be built for any platform that doesn't satisfy all of the
+constraints.
+<pre class="code">
+cc_library(
+    name = "foo_lib",
+    srcs = "foo_lib.cc",
+    target_compatible_with = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:windows",
+    ],
+)
+</pre>
+</p>
+
+<p>
+Use <code><a href="functions.html#select">select()</a></code> to express more
+complicated restrictions. For example, use an identity to implement basic OR
+logic. The following marks a library compatible with amd64 and arm64, but no
+other platforms.
+<pre class="code">
+cc_library(
+    name = "bar_lib",
+    srcs = "bar_lib.cc",
+    target_compatible_with = select({
+        "@platforms//cpu:x86_64": ["@platforms//cpu:x86_64"],
+        "@platforms//cpu:arm64": ["@platforms//cpu:arm64"],
+        "//conditions:default": ["//:not_compatible"],
+    ],
+)
+</pre>
+where <code>:not_compatible</code> is a user-defined
+<code><a href="platform.html#constraint_value">constraint_value</a></code> that
+no platform satisfies.
+</p>
+
+<p>
+Similarly, use a user-defined <code>:not_compatible<code> to enumerate
+constraints that a target is <i>not</i> compatible with.
+<pre class="code">
+cc_library(
+    name = "bar_lib",
+    srcs = "bar_lib.cc",
+    target_compatible_with = select({
+        "@platforms//os:osx": [":not_compatible"],
+        "@platforms//os:freebsd": [":not_compatible"],
+        "//conditions:default": [],
+    ],
+)
```

Comment:
Let's add a clarifying sentence saying explicitly what this produces: "This says bar_lib is compatible with everything except OSX and FreeBSD".

Reply:
Good idea. Done.

This now lives in https://docs.bazel.build/versions/master/platforms.html.